src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -84,17 +86,41 @@ Copyright (c) .NET Foundation. All rights reserved.
     <Delete Files="@(_LinkedResolvedFileToPublishCandidates)" />
     <ILLink AssemblyPaths="@(_ManagedAssembliesToLink)"
             ReferenceAssemblyPaths="@(ReferencePath)"
-            RootAssemblyNames="@(IntermediateAssembly->'%(Filename)')"
+            RootAssemblyNames="@(TrimmerRootAssembly)"
             RootDescriptorFiles="@(TrimmerRootDescriptor)"
             OutputDirectory="$(IntermediateLinkDir)"
-            ExtraArgs="-u copyused -c copyused -l none --skip-unresolved true $(_ExtraTrimmerArgs)"
+            ExtraArgs="$(_ExtraTrimmerArgs)"
             ToolExe="$(_DotNetHostFileName)"
             ToolPath="$(_DotNetHostDirectory)" />
 
      <Touch Files="$(_LinkSemaphore)" AlwaysCreate="true" />
 
   </Target>
 
+  <!--
+    ============================================================
+                     _SetILLinkDefaults
+
+    Set up the default options and inputs to ILLink.
+   -->
+   <Target Name="_SetILLinkDefaults"
+           DependsOnTargets="_ComputeManagedAssembliesToLink">
+
+    <NETSdkError Condition="'$(SelfContained)' != 'true'" ResourceName="ILLinkNotSupportedError" />
```

Comment:
Should we always cause an error on fw-dependent publishing? 

I understand that with the current defaults, there's no point in running the linker on fw-dependent builds. But it isn't an error.  Is there a way to override the default settings (as done in the tests) and run the linker on fw-dependent apps? 

Can you please add a comment clarifying why we disallow fw-dependent configuration?

Reply:
The currently-supported ways to override defaults are:
- adding roots via `TrimmerRootDescriptor` and `TrimmerRootAssembly`, which doesn't make it any more useful for portable apps, and
- using ILLink task directly, explicitly specifying the set of roots.

Somebody could do what the tests do and remove items from the defaults, but it requires depending on a private target.

In the future we may consider supporting something like `IsTrimmable` on `PackageReference`s, but until then I think we should keep the error - I can't think of a reasonable condition to put it under with the current override mechanisms. And someone who is willing to use the task directly will likely be able to work around the error as well.

I'll add a comment indicating as much.

Reply:
I also wanted to note more generally about overriding defaults - the out of band package had more high-level msbuild properties that mapped to different combinations of roots and linker actions. However it ended up with a bunch of logic in msbuild that wasn't tested and kept breaking. If we do introduce a high-level "mode" concept, we'll need to ensure test coverage of different modes, and probably make this concept known to illink itself.

Reply:
Thanks for the explanation @sbomer